### PR TITLE
Add new Swift Server Generator section to Getting Started guide

### PR DESCRIPTION
**Pull request by Joe-Brady**
On request of Helen Masters - a new section of documentation added.

All formatted to appear the same as the existing documentation.

One new addition to the menu structure.

### CODE REVIEW

File: en/starter/generator/getting_started.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+Through a set of simple steps using the Swift Server Generator tool, you’ve created a CoffeeShop model, specified its properties and then exposed it through REST.
+
+Models that are created by following this tutorial have a standard set of create, read, update and delete operations defined. See [Exposing Swift Server Generator models over REST](exposing_ssg_models_over_rest.md) for details. You can test any of the API endpoints by using **`curl`** commands. For example, to display all the model instances in the `swiftserver-getting-started` project, enter the following command (the `-k` option avoids certificate errors):
+
+    curl --request GET \\
+    --url 'https://localhost:8090/api/CoffeeShops' \\
+    --header 'accept: application/json' \\
+    --header 'content-type: application/json' \\
+    --header 'x-ibm-client-id: default' \\
+    --header 'x-ibm-client-secret: SECRET' -k
```
Comment (tunniclm) [inline en/starter/generator/getting_started.md]: This curl command is using the wrong protocol (should be http) and contains a lot of extra stuff that is not needed, especially the last two parameters which are API Connect specific and so not appropriate for this page. Should be:
```
curl http://localhost:8090/api/CoffeeShops
```

File: en/starter/generator/core_concepts.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+After you have generated an application and defined your models you will have a Kitura Swift application built by using the Swift Package Manager. [*Kitura*](http://www.kitura.io/) is an open source web framework for building Swift applications inspired by [*Express®*](http://expressjs.com/), a Node.js web framework. When you run your Kitura Swift application, it creates a Kitura server that handles REST/HTTP requests for each of the defined resources and their operations.
+
+A key powerful feature of Swift Server Generator is that when you define a model it automatically comes with a predefined REST API with a full set of create, read, update, and delete operations. This REST API is described in more detail in [Exposing Swift Server Generator models over REST](exposing_ssg_models_over_rest.md).
```
Comment (tunniclm) [inline en/starter/generator/core_concepts.md]: Broken link

File: en/starter/generator/core_concepts.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+You can create models by using the model generator, which creates a [Model definition JSON file](model_definition_json_file.md) that defines your model. By convention, this file is located in the project's `models` directory; for example, `models/customer.json`.
```
Comment (tunniclm) [inline en/starter/generator/core_concepts.md]: Broken link

File: en/starter/generator/installation.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+3.  To install this generator, run the following command:
+
+        $ npm install -g git+https://github.com/IBM-Swift/generator-swiftserver
+
+> ![info] Note: Swift Server Generator is supported only on the following operating systems:
+
+-   macOS capable of running Xcode 8.1
+-   Ubuntu Linux 14.04 LTS
+-   Ubuntu Linux 15.10
```
Comment (tunniclm) [inline en/starter/generator/installation.md]: I think 15.10 is out of support and you can't get newer Swift versions on that version any more.

File: en/starter/generator/command_line_tools.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+|---------------------------------  |------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **`yo swiftserver`**              | [Application generator](#application-generator) | Create a new Swift Server Generator application.                                                                                                                                                                                                                       |
+| **`yo create swiftserver:model`** | [Model generator](#model-generator)             | Add a new model to a Swift Server Generator application.                                                                                                                                                                                                               |
```
Comment (tunniclm) [inline en/starter/generator/command_line_tools.md]: The correct command is `yo swiftserver:model` the `create` part should not be there.

File: en/starter/generator/command_line_tools.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+| **`yo swiftserver:property`**     | [Property generator](#property-generator)       | Add a new property to an existing model.                                                                                                                                                                                                                               |
+| **`yo swiftserver:refresh`**      | Refresh generator                              | Generate OpenAPI (Swagger 2.0) and product definition YAML files, by convention named *`application`*`.yaml` and *`application`*`-product.yaml`. If the product file already exists it is not regenerated. The OpenAPI (Swagger 2.0) definition is always regenerated. |
```
Comment (tunniclm) [inline en/starter/generator/command_line_tools.md]: The description here is not correct and refers to API Connect-specific functionality.
Should be:
`Generate OpenAPI (Swagger 2.0) definition YAML file, by convention named after your application (e.g. appname.yaml). This file is regenerated each time the refresh generator runs.`

File: en/starter/generator/command_line_tools.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+**Options**
+
+    -h, --help
+
+Print the generator’s options and usage.
+
+    -n, --name
+
+Provide the application name.
```
Comment (tunniclm) [inline en/starter/generator/command_line_tools.md]: There is no `-n` or `--name` option. The name can be passed as an optional argument.

File: en/starter/generator/command_line_tools.md [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+
+**Interactive prompts**
+
+The generator prompts you for the following information:
+
+-   The name of the application to create, if you have not provided the application name as an option, as in the following example:
+
+        yo swiftserver --name bookshop
```
Comment (tunniclm) [inline en/starter/generator/command_line_tools.md]: Should be:
`yo swiftserver bookshop`

File: en/starter/generator/command_line_tools.md [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+
+-   The name of the directory to create the application in; this defaults to the application name you previously entered.
+
+-   Data store to use; this defaults to memory which is an in-memory data source that is built into Swift Server Generator and is suitable only for development and initial testing. If you want to connect your models to a real data source such as the Cloudant® database server, choose `cloudant` for your back-end data store. The tool prompts you for additional settings and automatically installs the appropriate Cloudant connector package by using the Swift Package Manager. If you want to switch between data stores see the [`config.json file`](config_json.md).
```
Comment (tunniclm) [inline en/starter/generator/command_line_tools.md]: Broken link

File: en/starter/generator/command_line_tools.md [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+
+The tool creates the standard Swift Server Generator application structure; for details, see [Swift Server Generator project layout reference](project_layout_reference.md).
```
Comment (tunniclm) [inline en/starter/generator/command_line_tools.md]: Broken link

File: en/starter/generator/command_line_tools.md [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+
+**Options**
+
+Print the generator’s options and usage:
+
+    -h, --help
+
+Provide the model name:
+
+    -n, --name
```
Comment (tunniclm) [inline en/starter/generator/command_line_tools.md]: There is no `-n` or `--name` option. The name can be passed as an optional argument.

File: en/starter/generator/command_line_tools.md [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+
+**Interactive prompts**
+
+The tool will prompt you for the following information:
+
+-   The name of the model, if you have not provided the model name as an option
+
+-   The custom plural form of the model name. By default, the tool pluralizes the model name by adding *s*.
+
+    > ![info] Note: All Swift Server Generator models are exposed over REST. A set of create, read, update, and delete (CRUD) operations are therefore available via REST endpoints; for details, see [Exposing Swift Server Generator models over REST](exposing_ssg_models_over_rest.md). Operations on all models use the model’s plural name in the URL; for example, `http://localhost:3000/api/bookshops`.
```
Comment (tunniclm) [inline en/starter/generator/command_line_tools.md]: Broken link

File: en/starter/generator/command_line_tools.md [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+
+The tool then invokes the [Property generator](#property-generator) and prompts you to enter model properties.
+
+**Output**
+
+The tool creates a new file that defines the model, /models/model-name.json; for details, see [Model definition JSON file](model_definition_json_file.md). It also generates an [*OpenAPI (Swagger 2.0) specification*](http://swagger.io/specification/) YAML file that describes and documents the REST APIs. By convention this file is located in the project’s definitions directory; for example, `definitions/appname.yaml`.
```
Comment (tunniclm) [inline en/starter/generator/command_line_tools.md]: Broken link

File: en/starter/generator/project_layout_reference.md [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+Swift Server Generator project files and directories are stored in a predefined structure within the application root directory.
+
+The standard project structure contains the following sub-directories within the application root directory:
+
+-   `Packages`: Swift packages that are defined as application dependencies.
+-   `Sources`: Swift source files.
+       - The `/<application_name>` subdirectory contains the `main.swift` file for the application.
+       - The `/Generated` subdirectory contains all generated Swift files, including those for models.
+-   `models`: All model JSON files.
+-   `definitions`: API and product definition YAML files.
```
Comment (tunniclm) [inline en/starter/generator/project_layout_reference.md]: `API and product definition YAML files` should be `API definition YAML files`

File: en/starter/generator/project_layout_reference.md [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+| **`/Sources` directory**            | Contains Swift application files. |
+| `/<application_name>` directory     | Contains the `main.swift` file for the application. |
+| `/Generated` directory              | Contains the generated Swift files for the application, including model Swift files. By convention, these files are called `<Model_name>.swift`; for example, a model called `customer` results in a Swift class file named `Customer.swift`. **Note:** Model names starting with a lower case alphabetic letter (as in the example above) are capitalized when they are converted to Swift classnames.                                                                                    |
+| **/models directory**               | Contains [model definition JSON files](model_definition_json_file.md), by convention named `<model_name>.json`; for example, `customer.json`. |
+| **/definitions directory**          | Contains API and product definition YAML files, by convention named `<application>.yaml` and `<application>-product.yaml`; for example `acmebank.yaml` and `acmebank-product.yaml`. |
```
Comment (tunniclm) [inline en/starter/generator/project_layout_reference.md]: The description here is not correct and refers to API Connect-specific functionality.
Should be:
`Contains API definition YAML file, by convention named after your application (e.g. acmebank.yaml).`

File: en/starter/generator/project_layout_reference.md [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+
+> ![info] Note: Model names are as you enter them in the model generator and are used in the application endpoints in this way. Model names are converted to Swift class names according to the [Swift specifications for Identifiers](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/LexicalStructure.html#//apple_ref/doc/uid/TP40014097-CH30-ID410). In addition, the convention that Swift class names start with an uppercase letter is applied, unless the model name starts with an underscore character (_).
```
Comment (tunniclm) [inline en/starter/generator/project_layout_reference.md]: I can't parse the grammar of the first sentence here. I think it should say something like:
`Model names entered into the model generator are used in a number of places in the generated project. They are used verbatim as filenames for the .json model files and as endpoint paths in the the application router and API definition. They are converted to Swift classnames in the generated Swift code using a simple transform that will attempt to make them comply to the [Swift specifications for Identifiers](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/LexicalStructure.html#//apple_ref/doc/uid/TP40014097-CH30-ID410). In addition, the convention that Swift class names start with an uppercase letter is applied, unless the model name starts with an underscore character (_).`

File: en/starter/generator/package_swift.md [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+For more information, see the [Package.swift documentation](https://swift.org/package-manager/).
+
+---
+
+## Example
+
+    import PackageDescription
+
+    let package = Package(
+      name: "GenKitExampleApplication",
```
Comment (tunniclm) [inline en/starter/generator/package_swift.md]: Should probably use `ExampleApplication` here for the name `GenKit` was an old codename for Swift Server Generator.

File: en/starter/generator/exposing_ssg_models_over_rest.md [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+	<h1>Exposing Swift Server Generator models over REST</h1>
+</div>
+
+Swift Server Generator models automatically have a standard set of HTTP endpoints that provide REST APIs for create, read, update and delete (CRUD) operations on model data.
+
+---
+
+## REST paths
+
+The REST APIs are mounted to the `plural` model name in the [Model definition JSON file](model_definition_json_file.md). For example, a model named `location` with plural `locations` has its REST API mounted on `/api/locations`.
```
Comment (tunniclm) [inline en/starter/generator/exposing_ssg_models_over_rest.md]: Broken link

File: en/starter/generator/package_swift.md [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+        .Package(url: "https://github.com/IBM-Swift/Kitura.git", majorVersion: 1, minor: 1),
+        .Package(url: "https://github.com/IBM-Swift/GeneratedSwiftServer.git", majorVersion: 0, minor: 3),
+        .Package(url: "https://github.com/IBM-Swift/HeliumLogger.git", majorVersion: 1, minor: 1),
+        .Package(url: "https://github.com/IBM-Swift/GeneratedSwiftServer-CloudantStore.git", majorVersion: 0, minor: 3),
+      ]
+    )
+
+    [info]: ../../../assets/info-blue.png
+    [tip]: ../../../assets/lightbulb-yellow.png
+    [warning]: ../../../assets/warning-red.png
```
Comment (tunniclm) [inline en/starter/generator/package_swift.md]: This metadata appears in the rendered page at the bottom of the example. Looks like some markup was not closed properly.

File: en/starter/generator/exposing_ssg_models_over_rest.md [hunk 20]
```diff
@@ -0,0 +1,85 @@
...
+For example:
+
+    {
+      "error": "Provided value (1234) for property 'type' has type (number) which is not compatible with the property type (string)"
+    }
+
+
+    [info]: ../../../assets/info-blue.png
+    [tip]: ../../../assets/lightbulb-yellow.png
+    [warning]: ../../../assets/warning-red.png
```
Comment (tunniclm) [inline en/starter/generator/exposing_ssg_models_over_rest.md]: This metadata appears in the rendered page at the bottom of the example. Looks like some markup was not closed properly.

**Comment by Joe-Brady** [comment]
@tunniclm 
Hi, thanks for reviewing.
When Helen gave me the documentation, I should have checked through all of the links as well as applied the formatting - didn't clock that, of course, her .md links should be .html.

This morning I will fix those links, and the parts that I understand, but there are some errors here that I think are best fixed either by the original author (Helen), or by someone else who understands the technical aspects.

File: en/starter/generator/installation.md (hunk 4 above)
Comment (lroseblade) [inline en/starter/generator/installation.md]: @tunniclm can you confirm this for @Joe-Brady and let him know what this should read (if any changes required)?

**Comment by lroseblade** [comment]
@Joe-Brady I think @tunniclm  has provided all the replacement text with the exception of the 15.10 support one, for which I've requested he confirms. So I think you have everything you need. Is there some information you are ambiguous about still?

Thanks

**Comment by Joe-Brady** [comment]
"There is no -n or --name option. The name can be passed as an optional argument." - what should the final text read instead?

File: en/starter/generator/getting_started.md [hunk 21]
```diff
@@ -0,0 +1,153 @@
...
+
+4.  Downloading and installing dependent Swift modules (as if you had manually run `swift build`).
+
+---
+
+## Create models
+
+Now that you’ve created the initial project, you’re going to create a *CoffeeShop* model that will automatically have REST API endpoints.
+
+Go into your new application directory, then run the Swift Server [model generator](http://loopback.io/doc/en/lb2/Model-generator):
```
Comment (Joe-Brady) [inline en/starter/generator/getting_started.md]: @helenmasters @tunniclm What should this link be?

File: en/starter/generator/getting_started.md (hunk 1 above)
Comment (Joe-Brady) [inline en/starter/generator/getting_started.md]: So replace lines 143-147 with just this single line?

File: en/starter/generator/command_line_tools.md [hunk 22]
```diff
@@ -0,0 +1,154 @@
...
+	<p>This article summarizes Swift Server Generator CLI tool commands.</p>
+</div>
+
+Two command-line tools are available for Swift Server Generator:
+
+-   [Yeoman](http://yeoman.io/) command-line tools: `yo swiftserver`.
+
+-   IBM API Connect developer toolkit CLI: `apic swiftserver`.
+
+You use these command line tools to create and configure Swift Server Generator applications. The table below describes using the Yeoman command-line options. If you are familiar with the IBM API Connect developer toolkit and prefer to use that you can refer to the Swift [Server Generator installation and command line options](https://www.ibm.com/support/knowledgecenter/en/SSMNED\_5.0.0/com.ibm.apic.toolkit.doc/capic\_swift\_overview.html).
```
Comment (Joe-Brady) [inline en/starter/generator/command_line_tools.md]: https://www.ibm.com/support/knowledgecenter/en/SSMNED_5.0.0/com.ibm.apic.toolkit.doc/capic_swift_overview.html

Looks like this is the intended link. Pandoc has added some escaping here again.

File: en/starter/generator/command_line_tools.md (hunk 7 above)
Comment (Joe-Brady) [inline en/starter/generator/command_line_tools.md]: `-n, --name` is mentioned twice. Should it be fixed for both occurrences? What should it read instead?

File: en/starter/generator/project_layout_reference.md (hunk 14 above)
Comment (Joe-Brady) [inline en/starter/generator/project_layout_reference.md]: This phrase is used twice. Should it be fixed in this way for both occurrences?

**Comment by lroseblade** [comment]
Replace -n or --name with 

**Name**

Set the application name

**Comment by Joe-Brady** [comment]
@lroseblade @tunniclm Hi Lloyd, Mike said to remove that little section entirely? I have done that already, so let me know if you've both decided that needs changing

**Comment by tunniclm** [comment]
Sorry, didn't get around to checking the modifications today and I have a bunch of other things I need to get done tonight. I should get a look tomorrow then hopefully we'll be in shape to merge.

**Comment by seabaylea** [comment]
Ping me on Slack when its merged and I'll tweet about it (with reference to the demo at Playgrounds).

**Comment by tunniclm** [comment]
lgtm

**Comment by tunniclm** [comment]
Note: these docs are consistent with the **master** branch of generator-swiftserver. They will need an update when we merge in all the new work we have been doing for the next release.

**Comment by ianpartridge** [comment]
Merged.

@seabaylea http://www.kitura.io/en/starter/generator.html :)

**Pull request merged**